csharp/examples/IoTDataPipelineExample/Model/SensorDataPipeline.cs

```diff
@@ -0,0 +1,225 @@
+                        if (!_colSubjectIdBuilderDict.ContainsKey(subjectId))
+                        {
+                            _colSubjectIdBuilderDict.Add(subjectId, new Int32Array.Builder());
+                            _colActivityLabelBuilderDict.Add(subjectId, new StringArray.Builder());
+                            _colTimestampBuilderDict.Add(subjectId, new TimestampArray.Builder());
+                            _colXAxisBuilderDict.Add(subjectId, new DoubleArray.Builder());
+                            _colYAxisBuilderDict.Add(subjectId, new DoubleArray.Builder());
+                            _colZAxisBuilderDict.Add(subjectId, new DoubleArray.Builder());
+                        }
+                        _colSubjectIdBuilderDict[subjectId].Append((int)item.subjectId);
+                        _colActivityLabelBuilderDict[subjectId].Append(item.activityLabel);
+                        _colTimestampBuilderDict[subjectId].Append((DateTimeOffset)item.timestamp);
+                        _colXAxisBuilderDict[subjectId].Append((double)item.x_Axis);
+                        _colYAxisBuilderDict[subjectId].Append((double)item.y_Axis);
+                        _colZAxisBuilderDict[subjectId].Append((double)item.z_Axis);
+                    }
+                }
+            }
+            Console.WriteLine($"Read from channel task finished!");
+        }
+
+        public async Task<string> PersistData()
+        {
+            List<RecordBatch> recordBatches = new List<RecordBatch>();
+
+            string currentPath = Directory.GetCurrentDirectory();
+            string arrowDataPath = Path.Combine(currentPath, "arrow");
+            if (!Directory.Exists(arrowDataPath))
+                Directory.CreateDirectory(arrowDataPath);
+
+            var memoryAllocator = new NativeMemoryAllocator(alignment: 64);
+
+            foreach (var key in _colSubjectIdBuilderDict.Keys)
+            {
+                var subjectId = key;
+
+                //var recordBatch = new RecordBatch.Builder(memoryAllocator)
+                //.Append("SubjectId", false, _colSubjectIdBuilderDict[subjectId].Build())
+                //.Append("ActivityLabel", false, _colActivityLabelBuilderDict[subjectId].Build())
+                //.Append("Timestamp", false, _colTimestampBuilderDict[subjectId].Build())
+                //.Append("XAxis", false, _colXAxisBuilderDict[subjectId].Build())
+                //.Append("YAxis", false, _colYAxisBuilderDict[subjectId].Build())
+                //.Append("ZAxis", false, _colZAxisBuilderDict[subjectId].Build())
+                //.Build();
+                Schema.Builder schemaBuilder = new Schema.Builder();
+
+                schemaBuilder.Field(new Field("SubjectId", Int32Type.Default, nullable: false));
+                schemaBuilder.Field(new Field("ActivityLabel", StringType.Default, nullable: false));
+                schemaBuilder.Field(new Field("Timestamp", Int64Type.Default, nullable: false));
```

Comment:
I think this should use the `TimestampType`.

```suggestion
                schemaBuilder.Field(new Field("Timestamp", TimestampType.Default, nullable: false));
```